Makefile

```diff
@@ -11,20 +24,37 @@ UNAME := $(shell uname)
 TOOLCHAIN ?= $(DEVKITARM)
 PREFIX ?= arm-none-eabi-
 
-export PATH := $(TOOLCHAIN)/bin:$(PATH)
+ifneq (,$(TOOLCHAIN))
+  ifneq (,$(wildcard $(TOOLCHAIN)/bin))
+    export PATH := $(TOOLCHAIN)/bin:$(PATH)
+  endif
+endif
 
 ifeq ($(UNAME),Darwin)
-	SHELL := env PATH=$(PATH) /bin/bash
+  SHELL := env PATH=$(PATH) /bin/bash
 endif
 
-CPP ?= $(PREFIX)cpp$(EXE)
+ifeq ($(MODERN),0)
+  CPP ?= $(PREFIX)cpp$(EXE)
+else
+  CPP := $(PREFIX)cpp$(EXE)
+endif
 AS := $(PREFIX)as$(EXE)
 LD := $(PREFIX)ld$(EXE)
 OBJCOPY := $(PREFIX)objcopy$(EXE)
 STRIP := $(PREFIX)strip$(EXE)
 
-CC1     := tools/agbcc/bin/agbcc$(EXE)
-CC1_OLD := tools/agbcc/bin/old_agbcc$(EXE)
+# note: the makefile must be set up so MODERNCC is never called
```

Comment:
Can you give more details on what this is accomplishing?

Reply:
Mainly refer to https://github.com/pret/pokeemerald

Reply:
What parts of that repo is this intended to imitate? Should I believe that this is a good thing solely because emerald does it?

Reply:
This change is mainly inspired by the Makefile section, which allows for the compilation of different versions of the code. Intermediate files and .o files are generated in separate directories to ensure that switching between different versions during compilation does not cause interference.